framework/src/window.cpp

```diff
@@ -513,94 +513,231 @@ namespace gvk
 	{
 		mPresentQueue = &aPresentQueue;
 	}
+	
+	void window::create_swap_chain(swapchain_creation_mode aCreationMode)
+	{
+		construct_swap_chain_creation_info(aCreationMode);
+
+		auto newSwapChain = context().device().createSwapchainKHRUnique(mSwapChainCreateInfo);
 
-	std::tuple<vk::UniqueSwapchainKHR, std::vector<avk::image_view>, avk::renderpass, std::vector<avk::framebuffer>> window::recreate_swap_chain()
+		auto lifetimeHandler = [this](vk::UniqueSwapchainKHR&& aOldResource) { this->handle_lifetime(std::move(aOldResource)); };
+
+		// assign the new swap chain instead of the old one, if one exists
+		avk::emplace_and_handle_previous(newSwapChain, std::move(mSwapChain), lifetimeHandler);
+
+		construct_backbuffers(aCreationMode);
+
+		// if we are creating a new swap chain from the ground up,
+		// set up fences and other basic initialization
+		// TODO: part of this section needs to be also executed if number of concurrent images change
```

Comment:
Is this `TODO` still unsolved? If so, let's create a new Issue.

Reply:
That is part of issue #67 already.

Reply:
This is already part of issue #67